### PR TITLE
Print NFT edition from master edition API added

### PR DESCRIPTION
**Pull request by rex-god**

### CODE REVIEW

File: src/modules/nft/components/mint-nft/mint-nft.service.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Injectable } from '@nestjs/common';
+import { Metaplex, keypairIdentity } from '@metaplex-foundation/js';
+import { PrintNftEditionDto } from './dto/mint-nft.dto';
+import { AccountUtils } from 'src/common/utils/account-utils';
+import { clusterApiUrl, Connection, PublicKey } from '@solana/web3.js';
+
+@Injectable()
+export class MintNftService {
+  async printNewEdition(printNftEditionDto: PrintNftEditionDto): Promise<any> {
+    const { network, private_key, mint_address, mint_authority, update_authority } = printNftEditionDto;
```
Comment (codinpsycho) [inline src/modules/nft/components/mint-nft/mint-nft.service.ts:10]: Need a try catch block to return proper errors back

File: src/modules/nft/components/mint-nft/open-api.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        example: {
+          success: true,
+          message: 'NFT Edition printed successfully',
+          result: {
+            metadataAccount: {
+              publicKey: '6QseZzjtXdA4i33o2eCTXFNE5vUHkLVc6Mn2exWgjrTo',
+              exists: true,
+              data: {
+                key: 4,
+                updateAuthority: 'BFefyp7jNF5Xq2A4JDLLFFGpxLq5oPEFKBAQ46KJHW2R',
```
Comment (codinpsycho) [inline src/modules/nft/components/mint-nft/open-api.ts:20]: Do we need to return all this? Only new token address should suffice I think

Comment (rex-god) [reply src/modules/nft/components/mint-nft/open-api.ts:20]: Ok will update.

File: src/modules/nft/components/mint-nft/mint-nft.controller.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  async update(@Body() printNftEditionDto: PrintNftEditionDto): Promise<any> {
+    try {
+      const res = await this.mintNftService.printNewEdition(printNftEditionDto);
+      return {
+        success: true,
+        message: 'NFT Edition printed successfully',
+        result: res,
+      };
+    } catch (err) {
+      throw new HttpException(err.message, HttpStatus.FORBIDDEN);
```
Comment (codinpsycho) [inline src/modules/nft/components/mint-nft/mint-nft.controller.ts:25]: Why a forbidden http status?

Comment (rex-god) [reply src/modules/nft/components/mint-nft/mint-nft.controller.ts:25]: Should it be `INTERNAL_SERVER_ERROR`?

Comment (codinpsycho) [reply src/modules/nft/components/mint-nft/mint-nft.controller.ts:25]: Yes

**Review by vgvishesh — APPROVED** [review]

**Pull request merged**